src/api/routes/bns/pricing.ts

```diff
@@ -0,0 +1,118 @@
+import * as express from 'express';
+import { RouterWithAsync, addAsync } from '@awaitjs/express';
+import { DataStore, DbBNSNamespace } from '../../../datastore/common';
+import {
+  makeRandomPrivKey,
+  getAddressFromPrivateKey,
+  TransactionVersion,
+  ReadOnlyFunctionOptions,
+  bufferCVFromString,
+  callReadOnlyFunction,
+  ClarityType,
+  tupleCV,
+  uintCV,
+  listCV,
+} from '@stacks/transactions';
+import { GetStacksTestnetNetwork } from './../../../bns-helpers';
+import {
+  BNSGetNamePriceResponse,
+  BNSGetNamespacePriceResponse,
+} from '@blockstack/stacks-blockchain-api-types';
+
+export function createBNSPriceRouter(db: DataStore): RouterWithAsync {
+  const router = addAsync(express.Router());
+  const stacksNetwork = GetStacksTestnetNetwork();
+
+  router.getAsync('/namespaces/:namespace', async (req, res) => {
+    const { namespace } = req.params;
+    if (namespace.length > 20) {
+      res.status(400).json({ error: 'Invalid namespace' });
+      return;
+    }
+    const randomPrivKey = makeRandomPrivKey();
+    const address = getAddressFromPrivateKey(randomPrivKey.data, TransactionVersion.Testnet);
+
+    const txOptions: ReadOnlyFunctionOptions = {
+      senderAddress: address,
+      contractAddress: 'ST000000000000000000002AMW42H',
```

Comment:
The `ChainID` needs passed to the `createBNSPriceRouter` function and used to specify this address (so both testnet and mainnet work)

Reply:
This is still missing testnet+mainnet support.